coalib/coala.py

```diff
@@ -59,11 +60,35 @@ def main(debug=False):
 
         if args.show_bears:
             from coalib.settings.ConfigurationGathering import (
-                get_filtered_bears)
-
-            local_bears, global_bears = get_filtered_bears(
-                args.filter_by_language, log_printer)
-
+                get_all_bears)
+            filtered_bears = get_all_bears(log_printer)
+            if args.filter_by_language:
+                # filter_by_language option should be deprecated soon.
+                # Since it is already covered by filter_by option.
+                log_printer.warn(
+                    "'--filter-by-language ...' option is deprecated. "
+                    "Use '--filter-by language ...' option instead")
+                args.filter_by = args.filter_by or []
+                args.filter_by.append(['language'] + args.filter_by_language)
+            if args.filter_by:
+                # Each iteration of the following loop applies
+                # filters one by one provided as arguments
+                for filter_by in args.filter_by:
+                    filter_by_x, *filter_args = filter_by
+                    if len(filter_args) == 0:
+                        continue
+                    try:
+                        filtered_bears = FilterHelper.get_filtered_bears(
+                            filter_by_x, filter_args, filtered_bears)
```

Comment:
This is just a design consideration, you don't have to implement it (it's actually not so performant, just for the concept).

Mathematically speaking, because we apply a filter on a set of bears, and especially in a sequence, the invocation order of filters may define different results. A filter usually applies to a single bear, so we don't need to pass a filter the complete set of bears to filter. To ensure that the order of invocation and actual set of bears is independent from the result, we could invoke a single function that just says "yeah this bear shall be taken into the results -> True" or "no the bear should be filtered out -> False".
This kind of principle btw exists in many languages (also in Python with list comprehensinos), they usually take in a lambda that defines when an object shall be kept or not with a boolean value.